.github/actions/flatpak-builder-lint/action.yaml

```diff
@@ -0,0 +1,63 @@
+name: Run flatpak-builder-lint
+description: Runs flatpak-builder-lint with exceptions
+inputs:
+  artifact:
+    description: Type of artifact to lint (builddir, repo, manifest)
+    required: true
+  path:
+    description: Path to flatpak-builder manifest or Flatpak build directory
+    required: true
+  validateToPublish:
+    description: If false, turns some errors to non-errors for non-publish workflow
+    required: false
+    default: true
+  workingDirectory:
+    description: Working directory to clone flatpak-builder-lint
+    required: false
+    default: ${{ github.workspace }}
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v3
+      with:
+        repository: flathub/flatpak-builder-lint
+        ref: v2.0.13
+        path: flatpak-builder-lint
+        set-safe-directory: ${{ inputs.workingDirectory }}
+
+    - name: Install Dependencies 🛍️
+      shell: bash
+      working-directory: ${{ inputs.workingDirectory }}
+      run: |
+        : Install Dependencies 🛍️
+        echo ::group::Install Dependencies
+        dnf install -y -q poetry jq
+        poetry -q -C flatpak-builder-lint install
+        echo ::endgroup::
+
+    - name: Run flatpak-builder-lint
+      id: result
+      shell: bash
+      working-directory: ${{ inputs.workingDirectory }}
+      run: |
+        : Run flatpak-builder-lint
+        ret=$(poetry -C flatpak-builder-lint run flatpak-builder-lint --exceptions ${{ inputs.artifact }} ${{ inputs.path }} || true)
```

Comment:
From the docs it seems like `flatpak-builder-lint` would fail if `inputs.artifact` contains an unsupported positional argument, but this code always falls back to a zero return value.

So I'd probably would rather check if the input contains either "builddir", "repo", or "manifest" and fail gracefully with an error message.

This would be good practise as you're designing a repository action and it should validate the input before handing it off to the tool.

Reply:
I improved the management of the exit code but I kept a step to check if `artifact` is valid to exit earlier.